### PR TITLE
chore: added isScaled option to BadgeNetwork to allow nonscaled usage

### LINKED ISSUE #10320: [chore] Add non-scaled option for BadgeNetwork
- With the new [NFT design](https://www.figma.com/design/TfVzSMJA8KwpWX8TTWQ2iO/Asset-list-and-details?node-id=1242-143952&m=dev), BadgeNetwork needs to have an option to not be scaled to content

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- This PR seeks to add `isScaled` option to BadgeNetwork to accommodate for the [new NFT design](https://www.figma.com/design/TfVzSMJA8KwpWX8TTWQ2iO/Asset-list-and-details?node-id=1242-143952&m=dev), which requires that the Badge does not scale with the NFT
- Also adds more tests and update the stories

## **Related issues**

Fixes: #10320 

## **Manual testing steps**

1. Go to storybook
2. Check out BadgeNetwork
3.

### **After**
https://github.com/user-attachments/assets/74e75f32-4c30-4c12-af81-44fe3be6e293

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Badges/Badge/variants/BadgeNetwork/BadgeNetwork.stories.tsx [hunk 1]
```diff
@@ -1,37 +1,52 @@
 /* eslint-disable react-native/no-inline-styles */
-/* eslint-disable react/display-name */
+// Third party dependencies
```
Comment (brianacnguyen) [inline app/component-library/components/Badges/Badge/variants/BadgeNetwork/BadgeNetwork.stories.tsx:2]: storybook cleanup

File: app/component-library/components/Badges/Badge/variants/BadgeNetwork/BadgeNetwork.styles.ts [hunk 2]
```diff
@@ -21,53 +21,43 @@ const styleSheet = (params: {
...
-   */
-  const badgeToContentScaleRatio = 0.5;
-  const borderWidthRatio = 1 / 16;
-  const borderWidth = Number(size) * borderWidthRatio;
-  const currentAvatarSizes = Object.values(AvatarSize).map((avatarSize) =>
-    Number(avatarSize),
-  );
-  const currentSmallestAvatarSize = Math.min(...currentAvatarSizes);
-  const currentLargestAvatarSize = Math.max(...currentAvatarSizes);
-  let scaleRatio = 1;
```
Comment (brianacnguyen) [inline app/component-library/components/Badges/Badge/variants/BadgeNetwork/BadgeNetwork.styles.ts:43]: Scaled logic moved to utils to be reusable in tests

**Review by Cal-L — APPROVED** [review]
Lgtm

**Review by georgewrmarshall — APPROVED** [review]
LGTM

**Pull request merged**